tests/s2_inference/test_encoding.py

```diff
@@ -149,3 +153,50 @@ def test_model_un_normalization(self):
 
                 assert abs(max_output_norm - 1) > eps, f"{name}, {sentence}"
                 assert abs(min_output_norm - 1) > eps, f"{name}, {sentence}"
+
+    def test_open_clip_vectorize(self):
+
+        names = ['open_clip/RN50/openai', 'open_clip/RN101/yfcc15m', 'open_clip/RN101-quickgelu/openai',
+                 'open_clip/ViT-g-14/laion2b_s12b_b42k']
+
+        sentences = ['hello', 'this is a test sentence. so is this.', ['hello', 'this is a test sentence. so is this.']]
+        device = 'cpu'
+        eps = 1e-9
+
+        for name in names:
+
+            model = _load_model(name, device=device)
+
+            for sentence in sentences:
+                output_v = vectorise(name, sentence, device, normalize_embeddings=True)
+
+                assert _check_output_type(output_v)
+
+                output_m = model.encode(sentence, normalize=True)
+
+                assert abs(torch.FloatTensor(output_m) - torch.FloatTensor(output_v)).sum() < eps
+
+    def test_open_clip_embedding_size(self):
+
+        # This is a full test as the list includes all the models. Note that the training dataset does not affect the
+        # embedding size.
+        names = ['open_clip/RN50/openai', 'open_clip/RN101/openai', 'open_clip/RN101-quickgelu/openai',
```

Comment:
same comment as above, just test two models

Reply:
(get ones that have different sizes)

Reply:
![image](https://user-images.githubusercontent.com/49334982/194793495-e01ad41d-f159-4135-a650-e4ac6e430b68.png)

resolved. test time reduced from 5 mins to  3mins now